src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -188,12 +188,16 @@ void LinearIR::debug_print(bool tds_as_pointers) const {
 
 void LinearIR::init_emitters(const std::shared_ptr<TargetMachine>& target) {
     for (auto& expr : m_expressions) {
-        if (!expr->get_emitter())
+        if (!expr->get_emitter()) {
             expr->m_emitter = target->get(expr->get_node()->get_type_info())(expr);
-    }
-    if (target->custom_segfault_detector) {
-        for (auto& expr : m_expressions) {
-            expr->m_emitter->set_custom_segfault_detector(true);
+#ifdef DEBUG_CAPS
+            if (target->custom_segfault_detector) {
+                if (!is_type<ov::snippets::op::LoopBase>(expr->get_node()) ||
+                    !is_type<ov::snippets::op::BroadcastMove>(expr->get_node())) {
+                        expr->m_emitter->set_custom_segfault_detector(true);
```

Comment:
What about other non-memory-access operations, like Add/Subtract/Hswish/Exp etc? Do we need to skip them here as well?

There is a dedicated class for memory access operations in snippets - [MemoryAccess](https://github.com/openvinotoolkit/openvino/blob/db4e3bf83d4ada08fdc8bb15dee613a067457a3b/src/common/snippets/include/snippets/op/memory_access.hpp#L22). All operations that read/write from/to memory are derived from this class, so I would expect that we `set_custom_segfault_detector` only if `ov::is_type<op::MemoryAccess>(expr->get_node())`. This would allow to significantly reduce custom segfault detector's overheads without missing any memory accesses.

Similar check [example](https://github.com/openvinotoolkit/openvino/blob/df03b0437a0d5213eb18a6408b490a4e3f1dbf81/src/common/snippets/src/lowered/pass/allocate_buffers.cpp#L41)

@v-Golubev, what do you think?

Reply:
Agree: setting a flag only for memory access ops is enough

Reply:
As discussed offline, only enable MemoryAccess ops and entrance Kernel op by default if enable this segfault detector capability. print_emitter_info() and build info is kept to future and extended use. done in [7c2d591](https://github.com/openvinotoolkit/openvino/pull/19935/commits/7c2d591eb939261b074e0071b76786c1f5c5e846). thanks guys!